File: Backend/src/models/client/subscription.model.js
const mongoose = require('mongoose');
const { toJSON, paginate } = require('../plugins');

const subscriptionSchema = mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
    },
    validityPeriod: {
      type: String,
      required: true,
    },
    description: {
      type: String,
      required: true,
    },
    /** Package price shown as "Amount" in the admin UI */
    amount: {
      type: Number,
    },
    /** Validity unit: DAY | WEEK | MONTH | YEAR (admin UI) */
    unit: {
      type: String,
      enum: ['DAY', 'WEEK', 'MONTH', 'YEAR'],
    },
    status: {
      type: Boolean,
      required: true,
      default: true,
    },
    clientId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Client',
    },
    zoneId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Zone',
    },
  },
  {
    timestamps: true,
  },
);

subscriptionSchema.index({ zoneId: 1, clientId: 1 }, { background: true });

// add plugin that converts mongoose to json
subscriptionSchema.plugin(toJSON);
subscriptionSchema.plugin(paginate);

/**
 * @typedef SubScription
 */
const SubScription = mongoose.model('SubScription', subscriptionSchema);

module.exports = SubScription;

File: Driver/lib/network/response_models/subscription_model.dart
import 'dart:ui';

import 'package:flutter/material.dart';

class SubscriptionModel {
  String? unit;
  String? name;
  String? validityPeriod;
  String? description;
  bool? status;
  int? amount;
  String? id, imageUrl;
  Color? color;
  MaterialColor? materialColor;

  SubscriptionModel(
      {this.unit,
      this.name,
      this.validityPeriod,
      this.description,
      this.status,
      this.amount,
      this.id,
      this.color,
      this.materialColor,
      this.imageUrl});

  SubscriptionModel.fromJson(Map<String, dynamic> json) {
    unit = json['unit'];
    name = json['name'];
    validityPeriod = json['validityPeriod'];
    description = json['description'];
    status = json['status'];
    amount = json['amount'];
    id = json['id'];
    if (json['color'] != null) {
      color = json['color'];
    }
    if (json['materialColor'] != null) {
      materialColor = json['materialColor'];
    }
    if (json['imageUrl'] != null) {
      imageUrl = json['imageUrl'];
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['unit'] = unit;
    data['name'] = name;
    data['validityPeriod'] = validityPeriod;
    data['description'] = description;
    data['status'] = status;
    data['amount'] = amount;
    data['id'] = id;
    if (color != null) {
      data['color'] = color;
    }
    if (materialColor != null) {
      data['materialColor'] = materialColor;
    }
    if (imageUrl != null) {
      data['imageUrl'] = imageUrl;
    }
    return data;
  }
}
